plugin/src/main/java/org/opensearch/ml/action/agents/TransportRegisterAgentAction.java

```diff
@@ -60,10 +66,42 @@ protected void doExecute(Task task, ActionRequest request, ActionListener<MLRegi
 
     private void registerAgent(MLAgent agent, ActionListener<MLRegisterAgentResponse> listener) {
         Instant now = Instant.now();
-        MLAgent mlAgent = agent.toBuilder().createdTime(now).lastUpdateTime(now).build();
+        String llmId;
+        if (agent.getType().equalsIgnoreCase(MLAgentType.CONVERSATIONAL.name()) && agent.getLlm() != null) {
+            llmId = agent.getLlm().getModelId();
+        } else {
+            llmId = null;
+        }
+        try (ThreadContext.StoredContext context = client.threadPool().getThreadContext().stashContext()) {
+            if (llmId != null) {
+                String[] excludes = new String[] { MLModel.MODEL_CONTENT_FIELD, MLModel.OLD_MODEL_CONTENT_FIELD };
+                mlModelManager.getModel(llmId, null, excludes, ActionListener.wrap(mlModel -> {
+                    processModelAndRegisterAgent(agent, mlModel.getIsHidden(), now, listener, context);
+                }, e -> {
+                    log.error("Failed to retrieve the ML model with ID: " + llmId, e);
+                    listener.onFailure(e);
+                }));
+            } else {
+                processModelAndRegisterAgent(agent, false, now, listener, context);
+            }
+
+        } catch (Exception e) {
+            log.error("Failed to get the model information " + llmId, e);
+            listener.onFailure(e);
+        }
+    }
+
+    private void processModelAndRegisterAgent(
+        MLAgent agent,
+        Boolean isHiddenModel,
+        Instant now,
+        ActionListener<MLRegisterAgentResponse> listener,
+        ThreadContext.StoredContext context
+    ) {
+        MLAgent mlAgent = agent.toBuilder().createdTime(now).lastUpdateTime(now).isHidden(isHiddenModel).build();
```

Comment:
`isHiddenModel` is for LLM , why we set it on Agent level?

Reply:
in my understanding if a agent is created with a hidden model, that agent will also be marked as hidden agent.

Reply:
It is better to rely on super admin to identify if an agent is hidden. It is possible that one of the tools make use of hidden model and based on current logic the agent will not be hidden.